js/web/web-backend.js

```diff
@@ -11,22 +11,23 @@ import snarkyJsWebSrc from 'string:../../../web_bindings/snarky_js_web.bc.js';
 export { initSnarkyJS, withThreadPool };
 
 let wasm = plonkWasm();
+globalThis.plonk_wasm = wasm;
+
 let init = wasm.default;
 /**
- * @type {Worker}
+ * @type {Promise<Worker>}
+ */
+let workerPromise;
+/**
+ * @type {number | undefined}
  */
-let worker;
+let numWorkers = undefined;
 
 async function initSnarkyJS() {
   const memory = allocateWasmMemoryForUserAgent(navigator.userAgent);
   await init(undefined, memory);
 
   let module = init.__wbindgen_wasm_module;
-  let numWorkers = await getEfficientNumWorkers();
```

Comment:
change 1 to this file: we don't call `getEfficientNumWorkers()` right away, because it's very expensive (inspects the device GPU via WebGL). Instead, we call it before we need workers the first time, which is likely a slow operation anyway

Reply:
Great optimization! Great for cases where we only need constants since that can all be done in JS, right?

Reply:
Yes!